examples/server/toolschemas/main.go

```diff
@@ -58,11 +58,11 @@ func (t *manualGreeter) greet(_ context.Context, req *mcp.CallToolRequest) (*mcp
 	if err := json.Unmarshal(req.Params.Arguments, &input); err != nil {
 		return errf("failed to unmarshal arguments: %v", err), nil
 	}
-	if err := t.inputSchema.Validate(input); err != nil {
+	if err := validateStruct(input, t.inputSchema); err != nil {
```

Comment:
This isn't right: once we re-marshal, we're no longer validating the original JSON.

For example, suppose the input contains `"foo": 0`, and the struct has `omitempty` for the `Foo` field, and `"foo"` is required. Then the original input is valid, but the re-marshalled input is not.

This is why we should expose helpers for this...

Reply:
Good point!